### PR TITLE
add QuantityType<Density> and QuantityType<Energy> and units for that

### LINKED ISSUE #5333: add density to QuantityType
While working on the onewire binding (#4287) I noticed that `QuantityType<Density>` is missing. This is useful for absolute humidity values (measured in g/cm^3). uom-se is unfortunately not provding the needed interfaces, so this cannot be easily added.

### LINKED ISSUE #5341: kWh are not recognized as Energy values
The recently introduced unit support has an "Energy" dimension defined, which uses by default the SI unit "J" (Joule).
In practice, the alternative "Wh" are used much more frequently and thus I would expect this unit to be usable for items with `Energy` dimension.

I defined an Item like this:
```
Number:Energy Energie			"Energie [%.1f kWh]"
```
Sending a value on the console
```
smarthome send Energie 123.456
```
results in
```
2018-04-03 13:27:05.491 [DEBUG] [.e.s.core.types.util.UnitUtils:108  ] - Unknown unit from pattern: kWh
```
and no unit being assigned to the item state.

As energy is likely the most common use case besides weather for units, I'd consider this a rather critical issue that needs to be addressed quickly. See also https://github.com/openhab/openhab2-addons/pull/2167#discussion_r178578851.

### PR DESCRIPTION
**Pull request by J-N-K**
fixes #5333
fixes #5341 

Signed-off-by: Jan N. Klug <jan.n.klug@rub.de>

### CODE REVIEW

**Comment by J-N-K** [comment]
Travis fails in unrelated JSON storage test.

**Comment by J-N-K** [comment]
retriggering build after fixes for tests have been merged

**Comment by maggu2810** [comment]
@htreu ping :wink:

File: bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/dimension/Density.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import javax.measure.Quantity;
+
+/**
+ * Define Density type (basic unit is kg/m^3)
+ *
+ * @author Jan N. Klug - initial contribution
+ *
+ */
+public interface Density extends Quantity<Density> {
+}
```
Comment (htreu) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/dimension/Density.java]: missing newline

**Comment by J-N-K** [comment]
done

**Comment by J-N-K** [comment]
Please do not merge. I'll integrate #5346. Otherwise we get merge conflicts.

**Comment by J-N-K** [comment]
If Travis is fine with these changes, they are ready for merge.

**Comment by J-N-K** [comment]
npm in PaperUI again...

**Pull request merged**